src/xref2/link.ml

```diff
@@ -205,6 +205,20 @@ and with_location :
     a Location_.with_location =
  fun fn env x -> { x with Location_.value = fn env x.Location_.value }
 
+(** Like [with_location] but allow to raise warnings. *)
+and with_location' :
```

Comment:
Is there any reason to keep the non-warning-raising `with_location` above? Why can't everything use this function instead?

Reply:
I preferred to use this function only when necessary to be able to change it easily later. It's adding yet an other "implicit" way of passing warnings and I'm not sure it's the right way.

Reply:
One function is easier to change than two, and having only one is definitely less error-prone. If you strongly feel we should keep both, then we should distinuish between them with something better than `'`.

Reply:
The part that must be able to change is `Lookup_failures` and "implicit" warnings added in `Error`, that use a global reference. This function is useful to mark where the side effects are done (instead of everywhere), to be changed easily later.
I changed its name.